core/consensus/grandpa/vote_graph/vote_graph_impl.cpp

```diff
@@ -443,80 +432,62 @@ namespace kagome::consensus::grandpa {
   }
 
   std::optional<BlockInfo> VoteGraphImpl::findAncestor(
-      const BlockInfo &block, const VoteGraph::Condition &condition) const {
-    // we store two nodes with an edge between them that is the canonical
-    // chain.
-    // the `node_key` always points to the ancestor node, and the
-    // `canonical_node` points to the higher node.
-    std::optional<Entry> canonical_node = std::nullopt;
-    BlockHash node_key;
-
-    auto nodes_opt = findContainingNodes(block);
-    if (not nodes_opt.has_value()) {
-      const Entry &entry = entries_.at(block.hash);
-      if (condition(entry.cumulative_vote)) {
-        return block;
-      }
+      VoteType vote_type,
+      const BlockInfo &block_arg,
+      const VoteGraph::Condition &condition) const {
+    for (auto block = block_arg;;) {
+      auto nodes_opt = findContainingNodes(block);
 
-      auto ancestor_it = std::rbegin(entry.ancestors);
-      if (ancestor_it == std::rend(entry.ancestors)) {
-        return std::nullopt;
-      }
+      if (not nodes_opt.has_value()) {
+        // The block has a vote-node in the graph.
+        const Entry &node = entries_.at(block.hash);
 
-      canonical_node = entry;
-      node_key = *ancestor_it;
-    } else {
-      auto &nodes = nodes_opt.value();
-      if (nodes.empty()) {
-        return std::nullopt;
-      }
+        // If the weight is sufficient, we are done.
```

Comment:
[minor] comment could be: If the condition is fulfilled...

Reply:
Thx